app/lib/features/segment_deletion_config.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Features
+  class SegmentDeletionConfig
+    def self.configure(config = {})
+      @configuration = new(config)
+    end
+
+    def self.configuration
+      (@configuration || configure)
+    end
+
+    def self.config
+      configuration.config
+    end
+
+    def self.enabled?
+      configuration.enabled?
+    end
+
+    def initialize(config = {})
+      format_config(config)
+      check_validity(config)
+    end
+
+    def self.valid_environment?(environment: Rails.env)
+      # CAUTION!:
+      # This is because this feature destroys real data unless the call to segment is stubbed and without real credentials in the config (test environment),
+      # so we prevent this feature to be used in preview or development thinking that be just checking around while actually destroying real data by accident
+      %w[production test].include?(environment)
+    end
+
+    def self.log_disabled_warning
+      warn_message = '[WARNING] Can\'t enable "deletion of users in segment.'
+      warn_message = if valid_environment?
+                       "#{warn_message} Please revise your config."
+                     else
+                       "#{warn_message} The #{Rails.env} environment is not valid for this feature."
+                     end
+      Rails.logger.warn warn_message
+    end
+
+    attr_reader :config, :enabled
+    alias enabled? enabled
+
+    private
+
+    def format_config(config)
+      config = (config || {}).symbolize_keys
+      @config = ActiveSupport::OrderedOptions.new.merge(config)
+    end
+
+    def check_validity(initial_config)
+      return (@enabled = false) if initial_config.blank?
+      valid_config = %i[email password].all? { |key| config.key?(key) }
```

Comment:
`enabled` should be only if email and password are both set.

I think something like:

```ruby
email, password = config.values_at(:email, :password)
@enabled = email.present? && password.present?
```

Reply:
I don't know if that's a better way or not, but it doesn't really matter much because I still have pending this:
![image](https://user-images.githubusercontent.com/11318903/54937233-4fef2380-4f24-11e9-8c26-5b467bb2ee6a.png)

Which changes a lot this part anyway 😂